[docs] Add documentation for cache configuration properties (#24623)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties.rst

  See description for ``shared-arbitrator.memory-pool-min-free-capacity``

Cache Properties
----------------

The configuration properties of AsyncDataCache and SSD cache are described here.

``async-cache-persistence-interval``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``0s``

>   The interval for persisting in-memory cache to SSD. Setting this config

Reviewer: ```suggestion
  The interval for persisting in-memory cache to SSD. Set this 
```

  to a non-zero value will activate periodic cache persistence.

``async-data-cache-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``boolean``
* **Default value:** ``true``

  In-memory cache.

``async-data-ssd-gb``
^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``0``

>   The size of the SSD.

Author: Unit is actually in GiB so I'll mention that, thanks!

``async-cache-ssd-path``
^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``/mnt/flash/async_cache.``
  

>   The directory that is mounted onto SSD.

Reviewer: ```suggestion
  The path of the directory that is mounted in the SSD.
```

Author: Changing to "The path of the directory that is mounted onto SSD." since the directory and anything within it is mounted onto the SSD and not within it.

``async-cache-max-ssd-write-ratio``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``double``
* **Default value:** ``0.7``
  

>   The max ratio of the number of in-memory cache entries being written to SSD cache 

Reviewer: ```suggestion
  The maximum ratio of the number of in-memory cache entries written to the SSD cache 
```

>   over the total number of cache entries. This is to control SSD cache write rate, 

Reviewer: ```suggestion
  over the total number of cache entries. Use this to control SSD cache write rate: 
```

>   and once the ratio exceeds this threshold, then we stop writing to SSD cache.

Reviewer: ```suggestion
  once the ratio exceeds this threshold then we stop writing to the SSD cache.
```

``async-cache-ssd-savable-ratio``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``double``
* **Default value:** ``0.125``

>   The min ratio of SSD savable (in-memory) cache space over the total cache space. 

Reviewer: ```suggestion
  
  The min ratio of SSD savable (in-memory) cache space over the total cache space. 
```

  Once the ratio exceeds this limit, we start writing SSD savable cache entries 
  into SSD cache.

``async-cache-min-ssd-savable-bytes``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``integer``
* **Default value:** ``16777216 (16 MB)``

>   Min SSD savable (in-memory) cache space to start writing SSD savable cache entries into SSD cache. 

Reviewer: ```suggestion
 
   Min SSD savable (in-memory) cache space to start writing SSD savable cache entries into SSD cache. 
```
Need a line space here, see screenshot for what this currently looks like in a local build. 

<img width="836" alt="Screenshot 2025-02-25 at 9 29 53 AM" src="https://github.com/user-attachments/assets/73e45e22-cf18-4c28-b33b-4fdceeed5eb5" />

>   NOTE: we only write to SSD cache when both async-cache-max-ssd-write-ratio and 

Reviewer: ```suggestion
  
  NOTE: we only write to SSD cache when both ``async-cache-max-ssd-write-ratio`` and 
```
Suggest line break to put "NOTE" as left-justified.

>   async-cache-ssd-savable-ratio conditions are satisfied.

Reviewer: ```suggestion
  ``async-cache-ssd-savable-ratio`` conditions are satisfied.
```

``async-cache-persistence-interval``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``0s``

>   The interval for persisting in-memory cache to SSD. Setting this config to a non-zero value will 

Reviewer: ```suggestion
  
  The interval for persisting in-memory cache to SSD. Set this configuration to a non-zero value to 
```

>   activate periodic cache persistence. The following time units are supported: 

Reviewer: ```suggestion
  activate periodic cache persistence. 
  
  The following time units are supported: 
```

  ns, us, ms, s, m, h, d.

``async-cache-ssd-disable-file-cow``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``bool``
* **Default value:** ``false``

>   In file systems, such as btrfs, supporting cow (copy on write), the ssd cache can use all ssd 

Reviewer: ```suggestion
  
  In file systems such as btrfs that support cow (copy on write), the SSD cache can use all of the SSD 
```

  space and stop working. To prevent that, use this option to disable cow for cache files.

``ssd-cache-checksum-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``bool``
* **Default value:** ``false``

>   When enabled, a CRC-based checksum is calculated for each cache entry written to SSD. 

Reviewer: ```suggestion
  
  When enabled, a CRC-based checksum is calculated for each cache entry written to SSD. 
```

  The checksum is stored in the next checkpoint file.

``ssd-cache-read-verification-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``bool``
* **Default value:** ``false``

>   When enabled, the checksum is recalculated and verified against the stored value when 

Reviewer: ```suggestion
  
  When enabled, the checksum is recalculated and verified against the stored value when 
```

  cache data is loaded from the SSD.

``cache.velox.ttl-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``bool``
* **Default value:** ``false``

>   Enable TTL for AsyncDataCache and SSD cache.

Reviewer: ```suggestion
  
  Enable TTL for AsyncDataCache and SSD cache.
```

``cache.velox.ttl-threshold``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``2d``

>   TTL duration for AsyncDataCache and SSD cache entries. The following time units 

Reviewer: ```suggestion
  
  TTL duration for AsyncDataCache and SSD cache entries. The following time units 
```

  are supported: ns, us, ms, s, m, h, d.

``cache.velox.ttl-check-interval``

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

* **Type:** ``boolean``
* **Default value:** ``true``

  In-memory cache.

``async-cache-ssd-gb``
^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``0``

  The size of the SSD. Unit is in GiB (gibibytes).

``async-cache-ssd-path``
^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``/mnt/flash/async_cache.``
  
  The path of the directory that is mounted onto the SSD.

``async-cache-max-ssd-write-ratio``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``double``
* **Default value:** ``0.7``
  
  The maximum ratio of the number of in-memory cache entries written to the SSD cache 
  over the total number of cache entries. Use this to control SSD cache write rate, 
  once the ratio exceeds this threshold then we stop writing to the SSD cache.

``async-cache-ssd-savable-ratio``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``double``
* **Default value:** ``0.125``
  
  The min ratio of SSD savable (in-memory) cache space over the total cache space.
  Once the ratio exceeds this limit, we start writing SSD savable cache entries 
  into SSD cache.

``async-cache-min-ssd-savable-bytes``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``integer``

> * **Default value:** ``16777216 (16 MB)``

Reviewer 2: Nit: this looks a bit funky with the conversion to MB within the same value being highlighted. Maybe mention this separately in the text that the default is 16MB?

> * **Default value:** ``16777216 (16 MB)``

Reviewer: ```suggestion
* **Default value:** ``16777216``
```
@czentgr, good catch! Please Add in the description for this entry a statement something like: 

```
The default value ``16777216`` is 16 MB.
```

  
  Min SSD savable (in-memory) cache space to start writing SSD savable cache entries into SSD cache.

> NOTE: we only write to SSD cache when both ``async-cache-max-ssd-write-ratio`` and 

Reviewer: ```suggestion
  NOTE: we only write to SSD cache when both ``async-cache-max-ssd-write-ratio`` and 
```
Indent the NOTE to be within the heading. See screenshot of current state in local doc 
<img width="825" alt="Screenshot 2025-03-03 at 9 41 55 AM" src="https://github.com/user-attachments/assets/e7bc991d-4731-47ef-a873-e14200e35171" />
build.

> ``async-cache-ssd-savable-ratio`` conditions are satisfied.

Reviewer: ```suggestion
  ``async-cache-ssd-savable-ratio`` conditions are satisfied.
```
Need to indent this line to be part of the NOTE in the description text.

``async-cache-persistence-interval``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``0s``
  
  The interval for persisting in-memory cache to SSD. Set this configuration to a non-zero value will 
  activate periodic cache persistence.
  
  The following time units are supported: 

>   ns, us, ms, s, m, h, d.

Reviewer: If you want this line to be left-justified, add a blank line above it. In my local doc build lines 360 and 361 are on the same line like so: 

"The following time units are supported:  ns, us, ms, s, m, h, d."

I'm only mentioning this if you intended line 361 to be left-justified. If you didn't, resolve this comment.

``async-cache-ssd-disable-file-cow``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``bool``
* **Default value:** ``false``
  

>   In file systems, such as btrfs, supporting cow (copy on write), the ssd cache can use all of the ssd

Reviewer: ```suggestion
  In file systems such as btrfs that support cow (copy on write), the SSD cache can use all of the SSD
```
Suggest minor edit for readability.

* **Type:** ``double``
* **Default value:** ``0.7``
  
  The maximum ratio of the number of in-memory cache entries written to the SSD cache 
  over the total number of cache entries. Use this to control SSD cache write rate, 
  once the ratio exceeds this threshold then we stop writing to the SSD cache.

``async-cache-ssd-savable-ratio``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``double``
* **Default value:** ``0.125``
  
  The min ratio of SSD savable (in-memory) cache space over the total cache space.
  Once the ratio exceeds this limit, we start writing SSD savable cache entries 
  into SSD cache.

``async-cache-min-ssd-savable-bytes``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``integer``
* **Default value:** ``16777216``
  
  Min SSD savable (in-memory) cache space to start writing SSD savable cache entries into SSD cache.

  The default value ``16777216`` is 16 MB.

  NOTE: we only write to SSD cache when both ``async-cache-max-ssd-write-ratio`` and 
  ``async-cache-ssd-savable-ratio`` conditions are satisfied.

``async-cache-persistence-interval``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* **Type:** ``string``
* **Default value:** ``0s``
  

>   The interval for persisting in-memory cache to SSD. Set this configuration to a non-zero value will 

Reviewer: ```suggestion
  The interval for persisting in-memory cache to SSD. Set this configuration to a non-zero value to 
```
